app/Libraries/Beatmapset/ChangeBeatmapOwners.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Libraries\Beatmapset;
+
+use App\Exceptions\InvariantException;
+use App\Jobs\Notifications\BeatmapOwnerChange;
+use App\Models\Beatmap;
+use App\Models\BeatmapOwner;
+use App\Models\BeatmapsetEvent;
+use App\Models\User;
+use Ds\Set;
+
+class ChangeBeatmapOwners
+{
+    private Set $userIds;
+
+    public function __construct(private Beatmap $beatmap, array $newUserIds, private User $source)
+    {
+        priv_check_user($source, 'BeatmapUpdateOwner', $beatmap->beatmapset)->ensureCan();
```

Comment:
seems like this should be part of controller (as it previously was) as indicated by the need to create additional user in test